x/distribution/simulation/invariants.go

```diff
@@ -45,20 +57,26 @@ func CanWithdrawInvariant(k distr.Keeper, sk types.StakingKeeper) sdk.Invariant
 		// cache, we don't want to write changes
 		ctx, _ = ctx.CacheContext()
 
-		// iterate over all bonded validators, withdraw commission
+		var remaining sdk.DecCoins
+
+		// iterate over all validators
 		sk.IterateValidators(ctx, func(_ int64, val sdk.Validator) (stop bool) {
 			_ = k.WithdrawValidatorCommission(ctx, val.GetOperator())
+			// TODO fetch delegations just for the validator, requires sdk.ValidatorSet change
+			// iterate over all current delegations, withdraw rewards
+			dels := sk.GetAllSDKDelegations(ctx)
+			for _, delegation := range dels {
+				if delegation.GetValidatorAddr().String() == val.GetOperator().String() {
+					_ = k.WithdrawDelegationRewards(ctx, delegation.GetDelegatorAddr(), delegation.GetValidatorAddr())
```

Comment:
Should be not be checking this error?

Reply:
...... if not a comment as to that reason in the code would be useful

Reply:
sure, checked now